File: task1.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace ConsoleApp1
{
    internal class Program
    {
        static void Main(string[] args)
        {

            int number =16;
            int foundNumber = 0;

            for ( int i = number; i > 1; i= i / 2)
            {
                foundNumber = i;
                
            }
            if (foundNumber /2 == 1)
            {
                Console.WriteLine("it is power of 2");

            }
            else
            {
                Console.WriteLine("it is not power of 2");
            }
        }
    }
}

File: task1.js
//1 - Verilmiş ədədin ikinin qüvvəti olub olmadığını yoxlayan alqoritm yazın

let number = 4;
let foundNumber = 0;

for (let i = number; i>1; i=i/2) {
    foundNumber=i;
}

if (foundNumber/2==1) {
    console.log("it is power of 2");
    
}
else{
    console.log("it is not power of 2");
}